src/routes/__layout.svelte

```diff
@@ -1,5 +1,16 @@
-<script>
+<script lang="ts">
+  import {navigating} from '$app/stores';
+  import NProgress from 'nprogress';
+
+  $: if ($navigating) {
+    NProgress.start();
+  } else {
+    NProgress.done();
+  }
+
   import "../app.css";
 </script>
 
-<slot />
+<template>
+  <slot />
+</template>
```

Comment:
What does the `<template>` do? Not familiar with this.

Reply:
It allows you to use HTML preprocessors. But if left blank it will just act as a seperation between the script and style tags. Much like Vue.